### ISSUE TITLE
Please allow the KnativeClient to interact with resources from serving.knative.dev via the v1 binding classes

### ISSUE THREAD

**Issue by FWiesner** [fabric8io/kubernetes-client#2003]
`knativeClient.services()` still is based on `v1alpha1`

**Comment by rohanKanojia** [comment]
Thanks a lot for proving feedback for our knative extensions. These seem to be getting a lot of traction recently and are moving so fast :laughing: . We'll update these before next to next release.

**Comment by rohanKanojia** [comment]
@FWiesner : Do you think this issue is closed by #2016?

**Comment by FWiesner** [comment]
Well, I was able today to use the v1 `Service uilder`, but still I got the warning about an unstable API `v1alpha1` (Could also be beta- I sit in a car to Bangalore airport and cannot check). I have to review my case again, but, AFAIK v1 is enabled to author code now, but is not used in the communication with the API Server

**Comment by rohanKanojia** [comment]
@FWiesner : polite ping, did you get some time to check this issue again? I ran `ServiceCrudTest` and it seems like we're using `serving.knative.dev/v1` endpoints while hitting API server:
```
~/go/src/github.com/fabric8io/kubernetes-client/extensions/knative/tests : $ mvn test -Dtest="io.fabric8.knative.test.crud.ServiceCrudTest"
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 1 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 1 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO] 
[INFO] ----------------------< io.fabric8:knative-tests >----------------------
[INFO] Building Fabric8 :: Knative :: Tests 4.9-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- sundr-maven-plugin:0.21.0:generate-bom (default) @ knative-tests ---
[INFO] 
[INFO] --- build-helper-maven-plugin:3.1.0:parse-version (parse-version) @ knative-tests ---
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.4:prepare-agent (default-prepare-agent) @ knative-tests ---
[INFO] argLine set to -javaagent:/home/rohaan/.m2/repository/org/jacoco/org.jacoco.agent/0.8.4/org.jacoco.agent-0.8.4-runtime.jar=destfile=/home/rohaan/go/src/github.com/fabric8io/kubernetes-client/extensions/knative/tests/target/jacoco.exec
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:resources (default-resources) @ knative-tests ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/rohaan/go/src/github.com/fabric8io/kubernetes-client/extensions/knative/tests/src/main/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ knative-tests ---
[INFO] No sources to compile
[INFO] 
[INFO] --- jandex-maven-plugin:1.0.7:jandex (make-index) @ knative-tests ---
[INFO] [SKIP] Cannot process fileset in directory: /home/rohaan/go/src/github.com/fabric8io/kubernetes-client/extensions/knative/tests/target/classes. Directory does not exist!
[INFO] 
[INFO] --- maven-resources-plugin:3.1.0:testResources (default-testResources) @ knative-tests ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /home/rohaan/go/src/github.com/fabric8io/kubernetes-client/extensions/knative/tests/src/test/resources
[INFO] 
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ knative-tests ---
[INFO] Nothing to compile - all classes are up to date
[INFO] 
[INFO] --- maven-surefire-plugin:3.0.0-M4:test (default-test) @ knative-tests ---
[INFO] 
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running io.fabric8.knative.test.crud.ServiceCrudTest
Apr 02, 2020 12:22:52 PM okhttp3.mockwebserver.MockWebServer$2 execute
INFO: MockWebServer[54045] starting to accept connections
Apr 02, 2020 12:22:52 PM okhttp3.mockwebserver.MockWebServer$3 processOneRequest
INFO: MockWebServer[54045] received request: GET /apis/serving.knative.dev/v1/namespaces/ns1/services HTTP/1.1 and responded: HTTP/1.1 200 OK
Apr 02, 2020 12:22:52 PM okhttp3.mockwebserver.MockWebServer$2 acceptConnections
INFO: MockWebServer[54045] done accepting connections: Socket closed
Apr 02, 2020 12:22:53 PM okhttp3.mockwebserver.MockWebServer$2 execute
INFO: MockWebServer[59233] starting to accept connections
Apr 02, 2020 12:22:53 PM okhttp3.mockwebserver.MockWebServer$3 processOneRequest
INFO: MockWebServer[59233] received request: POST /apis/serving.knative.dev/v1/namespaces/ns2/services HTTP/1.1 and responded: HTTP/1.1 202 OK
Apr 02, 2020 12:22:53 PM okhttp3.mockwebserver.MockWebServer$3 processOneRequest
INFO: MockWebServer[59233] received request: GET /apis/serving.knative.dev/v1/namespaces/ns2/services HTTP/1.1 and responded: HTTP/1.1 200 OK
Apr 02, 2020 12:22:53 PM okhttp3.mockwebserver.MockWebServer$3 processOneRequest
INFO: MockWebServer[59233] received request: GET /apis/serving.knative.dev/v1/namespaces/ns2/services/service2 HTTP/1.1 and responded: HTTP/1.1 200 OK
Apr 02, 2020 12:22:53 PM okhttp3.mockwebserver.MockWebServer$2 acceptConnections
INFO: MockWebServer[59233] done accepting connections: Socket closed
Apr 02, 2020 12:22:53 PM okhttp3.mockwebserver.MockWebServer$2 execute
INFO: MockWebServer[58085] starting to accept connections
Apr 02, 2020 12:22:53 PM okhttp3.mockwebserver.MockWebServer$3 processOneRequest
INFO: MockWebServer[58085] received request: POST /apis/serving.knative.dev/v1/namespaces/ns3/services HTTP/1.1 and responded: HTTP/1.1 202 OK
Apr 02, 2020 12:22:53 PM okhttp3.mockwebserver.MockWebServer$3 processOneRequest
INFO: MockWebServer[58085] received request: DELETE /apis/serving.knative.dev/v1/namespaces/ns3/services/service3 HTTP/1.1 and responded: HTTP/1.1 200 OK
Apr 02, 2020 12:22:53 PM okhttp3.mockwebserver.MockWebServer$2 acceptConnections
INFO: MockWebServer[58085] done accepting connections: Socket closed
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.293 s - in io.fabric8.knative.test.crud.ServiceCrudTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  6.481 s
[INFO] Finished at: 2020-04-02T12:22:53+05:30
[INFO] ------------------------------------------------------------------------
~/go/src/github.com/fabric8io/kubernetes-client/extensions/knative/tests : $ 
```
Shall I close this issue? Do you think there is something pending related to this issue? Would appreciate if you could find time to provide some feedback related to this.

**Comment by FWiesner** [comment]
yes, please close it

Am Do., 2. Apr. 2020 um 08:57 Uhr schrieb Rohan Kumar <
notifications@github.com>:

**Comment by rozuhlinh** [comment]
> On Apr 2, 2020, at 3:01 AM, Rohan Kumar <notifications@github.com> wrote:
> 

﻿
Closed #2003.
